### PR TITLE
feat: add upcloud.com cloud support

### PR DESCRIPTION
**Pull request by sergelogvinov**
* cloud-init for upcloud.com
* ipv4/v6 support
* images format upcloud-$arch.raw.xz

## Acceptance

Please use the following checklist:

- [ ] you linked an issue (if applicable)
- [ ] you included tests (if applicable)
- [ ] you ran conformance (`make conformance`)
- [x] you formatted your code (`make fmt`)
- [x] you linted your code (`make lint`)
- [ ] you generated documentation (`make docs`)
- [ ] you ran unit-tests (`make unit-tests`)

> See `make help` for a description of the available targets.

### CODE REVIEW

**Review by smira — APPROVED** [review]

File: internal/app/machined/pkg/runtime/v1alpha1/platform/upcloud/upcloud_test.go [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+	suite.Require().NoError(err)
+	suite.Assert().Equal(machineConfig, result)
+}
+
+func TestConfigSuite(t *testing.T) {
+	suite.Run(t, new(ConfigSuite))
+}
+
+// http://169.254.169.254/metadata/v1.json
```
Comment (andrewrynhard) [inline internal/app/machined/pkg/runtime/v1alpha1/platform/upcloud/upcloud_test.go:141]: Do we need this comment?

**Review by andrewrynhard — APPROVED** [review]

File: internal/app/machined/pkg/runtime/v1alpha1/platform/upcloud/upcloud_test.go (hunk 1 above)
Comment (smira) [reply internal/app/machined/pkg/runtime/v1alpha1/platform/upcloud/upcloud_test.go:141]: yep, feels like duplicate, let me remove this

**Review by smira — APPROVED** [review]

**Review by andrewrynhard — APPROVED** [review]

**Pull request merged**